Add GitHub CI job to check Markdown links during PRs

joelhans opened the pull request:
<!--
Describe the change in summary section, including rationale and design decisions.
Include "Fixes #nnn" if you are fixing an existing issue.

In "Component Name" section write which component is changed in this PR. This
will help us review your PR quicker.

In "Test Plan" provide enough detail on how you plan to test this PR so that a reviewer can validate your tests. If our CI covers sufficient tests, then state which tests cover the change.

If you have more information you want to add, write them in "Additional
Information" section. This is usually used to help others understand your
motivation behind this change. A step-by-step reproduction of the problem is
helpful if there is no related issue.
-->

##### Summary

Third step of #8990. This PR adds a GitHub CI workflow to check changed Markdown files for broken links. This check will ensure we don't merge PRs that contain broken links.

If we can implement #8991, #8994, and this PR, we should be ready to completely disconnect the old Netlify CI workflows.

I'm relatively new to GitHub Actions, so apologies in advance...

##### Component Name

.github
docs

##### Test Plan

<!---
Provide enough detail so that your reviewer can understand which test-cases you
have covered, and recreate them if necessary. If sufficient tests are covered
by our CI, then state which tests cover the change.
-->

This PR contains a broken link so reviewers can check its behavior and output. I'll remove it before merging.

##### Additional Information

prologic commented on .github/workflows/check-links.yml:
```diff
@@ -0,0 +1,19 @@
+---
+name: Check Markdown links
```
```suggestion
name: Docs
```

prologic commented on .github/workflows/check-links.yml:
```diff
@@ -0,0 +1,19 @@
+---
+name: Check Markdown links
+on:
+  push:
+    branches:
+      - master
+  pull_request:
```
```suggestion
  push:
    branches:
      - master
    paths:
      - '*.md'
  pull_request:
    paths:
      - '*.md'
```

prologic commented on .github/workflows/check-links.yml:
```diff
@@ -0,0 +1,19 @@
+      - master
+  pull_request:
+jobs:
+  markdown-link-check:
+    runs-on: ubuntu-latest
```
```suggestion
  markdown-link-check:
    name: Broken Links
    runs-on: ubuntu-latest
```

prologic commented on .github/workflows/check-links.yml:
```diff
@@ -0,0 +1,19 @@
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@master
```
```suggestion
        uses: actions/checkout@v2
```

prologic commented on registry/README.md:
```diff
@@ -15,6 +15,8 @@ features, you need to Sign In to Netdata cloud. By signing in, you opt in to let
 information described [here](#what-data-does-the-registry-store). You can still get the node menu, but not the cloud
 features, if you [run your own registry](#run-your-own-registry).
 
+[This is a broken link](/broken/link/README.md).
+
```
I _assume_ you will remove this later in this PR once reviewed/approved :)

joelhans replied:
You read my mind :smile:

joelhans commented:
@prologic I addressed your comments and removed the broken link. Thanks for the quick feedback!

Here's what the error output looks like, for reference:

```
 =========================> MARKDOWN LINK CHECK <=========================


FILE: registry/README.md
[✖] /broken/link/README.md → Status: 400 [Error: ENOENT: no such file or directory, access '/broken/link/README.md'] {

=========================================================================
```

prologic commented:
> @prologic I addressed your comments and removed the broken link. Thanks for the quick feedback!

No worries! Sorry for the slow follow up ;/ I've been swamped with large volumes of notifs and some hard focus time to get things done. This happens from time-to-time :) bare with me! I'll re-review this today!

joelhans commented:
@prologic Thank you very much! Let's take a day or two to make sure this definitely works as expected on new PRs and then we can move ahead with #8783.

prologic commented:
> @prologic Thank you very much! Let's take a day or two to make sure this definitely works as expected on new PRs and then we can move ahead with #8783.

Sure! Sounds good to me. Please remind me when its time and just hit me up on Slack!
